src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -491,10 +491,8 @@ internal static int GetDefaultWarningLevel(DiagnosticSeverity severity)
                     return 0;
 
                 case DiagnosticSeverity.Warning:
-                    return 1;
-
                 default:
```

Comment:
>default: [](start = 16, length = 8)

It is not clear why 1 is better value to return for the default case.  Why not 0?

Reply:
Consider adding a comment.

Reply:
The main requirement here is `DiagnosticSeverity.Warning` must be level 1. Otherwise if you try to set `/warn:0 /warnaserror` then items declared with severity Warning in a rule set file will cause the build to fail. This would be a behavior regression.

It's unclear whether `Info` and `Hidden` could be applied with level 0, and if so, what advantage that would have.